aws/validators.go

```diff
@@ -2466,3 +2466,37 @@ func validateNestedExactlyOneOf(m map[string]interface{}, valid []string) error
 	}
 	return nil
 }
+
+func MapLenBetween(min, max int) schema.SchemaValidateFunc {
+	return func(i interface{}, k string) (warnings []string, errors []error) {
+		m, ok := i.(map[string]interface{})
+		if !ok {
+			errors = append(errors, fmt.Errorf("expected type of %s to be map", k))
+			return warnings, errors
+		}
+
+		if len(m) < min || len(m) > max {
+			errors = append(errors, fmt.Errorf("expected length of %s to be in the range (%d - %d), got %d", k, min, max, len(m)))
+		}
+
+		return warnings, errors
+	}
+}
```

Comment:
In this case, I would just check the maximum size, since there isn't really a lower limit. You could call it `MapMaxLen()` or match it with the `MaxItems` field for `TypeList` and `TypeSet` and call it `MapMaxItems()`.

This would be a great addition to the [Terraform Plugin SDK](https://github.com/hashicorp/terraform-plugin-sdk)